src/analyses/call_graph.cpp

```diff
@@ -443,3 +491,91 @@ void find_leaves_below_function(
   std::unordered_set<irep_idt, dstring_hash> to_avoid;
   find_leaves_below_function(call_graph, function, to_avoid, output);
 }
+
+/*******************************************************************\
+
+Function: find_direct_or_indirect_callees_of_function
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+void find_direct_or_indirect_callees_of_function(
+  const call_grapht &call_graph,
+  const irep_idt &function,
+  std::unordered_set<irep_idt, dstring_hash> &output)
+{
+  std::unordered_set<irep_idt, dstring_hash> leaves;
+  find_leaves_below_function(call_graph, function, output, leaves);
+  output.insert(leaves.cbegin(), leaves.cend());
+}
+
+/*******************************************************************\
+
+Function: find_nearest_common_callees
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+void find_nearest_common_callees(
```

Comment:
I agree with that if these functions provides convenience functions for manipulating the object through the public interface. There might be more reasons for doing that, e.g. avoiding clutter, reducing class size, not being allowed to change the class, etc. Then these functions should even go into a separate "utility" module. 
One could argue that this is mixing procedural and object-oriented styles, or that functions floating around in the root namespace are bad (which could be improved by actually using namespaces ;) , which is, however, a global change to be done to the whole code base).